doc/index.rst

```diff
@@ -95,3 +95,20 @@ package in your environment in a format that can be used in ``requirements.txt``
 
 ``conda env export -n <env-name>`` is the equivalent for anaconda's environment.yml
 file.
+
+If using a Dockerfile, source a reproducible, tagged image
```

Comment:
Let's move all of this over to the `dockerfile.rst` document. My understanding is that we are trying to guide folks to using Dockerfiles as a choice in a very limited use case (i.e. when pip and conda can't get you what is needed). I think it's fine to have a short paragraph or a few bullets here, but the current text is mixing implementation with general tips for reproducibility.

Reply:
+1